src/codegen.cpp

```diff
@@ -3450,6 +3451,29 @@ static void finalize_gc_frame(Function *F)
         ptlsStates->setCalledFunction(getter);
         ptlsStates->setAttributes(jltls_states_func->getAttributes());
     }
+    else if (jl_tls_offset != -1) {
+#ifdef LLVM37
+        const char *asm_str = nullptr;
+#  if defined(_CPU_X86_64_)
+        asm_str = "movq %fs:0, $0";
+#  elif defined(_CPU_X86_)
+        asm_str = "movl %gs:0, $0";
+#  elif defined(_CPU_AARCH64_)
+        asm_str = "mrs $0, tpidr_el0";
+#  endif
```

Comment:
I think what we want is a description of what it *does* do on other architectures. Something like "for the 3 supported architectures above, load tls pointer from its known offset. on all others, fall back to calling jl_get_ptls_states" (I think that's how it works?)